torch/_dynamo/variables/tensor.py

```diff
@@ -681,13 +684,49 @@ def redistribute_fn_with_prim_types(x):
                 mutable_local=variables.base.MutableLocal(),
                 **options,
             )
+
             if not self.source:
                 # Intermediary
-                unimplemented("Intermediary tensors with registered hooks - NYI")
-            else:
-                assert (
-                    fn_var.source
-                ), "Unreachable - See unimplemented for lambdas above"
+                src = fn_var.source
+                if (
+                    not src
+                    and isinstance(fn_var, variables.functions.FunctoolsPartialVariable)
+                    and fn_var.func.source
+                ):
+                    src = fn_var.func.source
+
+                if src:
+                    tx.output.guards.add(src.make_guard(GuardBuilder.ID_MATCH))
```

Comment:
Don't you want to error/graph break if we haven't computed the src by this point?

Reply:
yea, I think so